src/server/location.rs

```diff
@@ -265,7 +279,7 @@ impl Location {
             traits: None }
         }
         */
-        let mut result = LocationResult::default();
+        let mut result = self.default_loc.clone();
```

Comment:
AFAICT this doesn't plug the defaults into `LocationResult` -- this only plugs it into the mmdb code.

Reply:
Well, we're doing a bit of a shortcut here. I did see that we're not doing the same for the Google Header, so modified that. 

Also moved the location fixup code out of settings because it really should be part of the location impl.